operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,304 @@
+	// The preemption algorithm will only use the next strategy in the list if the
+	// incoming workload (preemptor) doesn't fit after using the previous strategies.
+	// Possible values are:
+	// - LessThanOrEqualToFinalShare: Only preempt a workload if the share of the preemptor CQ
+	//   with the preemptor workload is less than or equal to the share of the preemptee CQ
+	//   without the workload to be preempted.
+	//   This strategy might favor preemption of smaller workloads in the preemptee CQ,
+	//   regardless of priority or start time, in an effort to keep the share of the CQ
+	//   as high as possible.
+	// - LessThanInitialShare: Only preempt a workload if the share of the preemptor CQ
+	//   with the incoming workload is strictly less than the share of the preemptee CQ.
+	//   This strategy doesn't depend on the share usage of the workload being preempted.
+	//   As a result, the strategy chooses to preempt workloads with the lowest priority and
+	//   newest start time first.
+	// The default strategy is ["LessThanOrEqualToFinalShare", "LessThanInitialShare"].
+	// +optional
+	// +kubebuilder:validation:MaxItems=2
+	PreemptionStrategies []PreemptionStrategy `json:"preemptionStrategies,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=batch/job;ray.io/rayjob;ray.io/raycluster;jobset.x-k8s.io/jobset;kubeflow.org/mpijob;kubeflow.org/paddlejob;kubeflow.org/pytorchjob;kubeflow.org/tfjob;kubeflow.org/xgboostjob;workload.codeflare.dev/appwrapper;pod;deployment;statefulset;leaderworkerset.x-k8s.io/leaderworkerset
+type KueueIntegrations string
+
+const (
+	BatchJob        KueueIntegrations = "batch/job"
+	RayJob          KueueIntegrations = "ray.io/rayjob"
+	RayCluster      KueueIntegrations = "ray.io/raycluster"
+	JobSet          KueueIntegrations = "jobset.x-k8s.io/jobset"
+	MPIJob          KueueIntegrations = "kubeflow.org/mpijob"
+	PaddeJob        KueueIntegrations = "kubeflow.org/paddlejob"
+	PyTorchJob      KueueIntegrations = "kubeflow.org/pytorchjob"
+	TfJob           KueueIntegrations = "kubeflow.org/tfjob"
+	XGBoostJob      KueueIntegrations = "kubeflow.org/xgboostjob"
+	AppWrappers     KueueIntegrations = "workload.codeflare.dev/appwrapper"
+	Pod             KueueIntegrations = "pod"
+	Deployment      KueueIntegrations = "deployment"
+	Statefulset     KueueIntegrations = "statefulset"
+	LeaderWorkerSet KueueIntegrations = "leaderworkerset.x-k8s.io/leaderworkerset"
+)
+
+// This is the GVK for an external framework.
+// Controller runtime requires this in this format
+// for api discoverability.
+type ExternalFramework struct {
+	// group of externalFramework
+	// +kubebuilder:validation:MaxLength=256
+	// +required
+	Group string `json:"group"`
+	// resourceType of external framework
+	// this is the same as Kind in the GVK settings
+	// +required
+	// +kubebuilder:validation:MaxLength=256
+	ResourceType string `json:"resourceType"`
+	// version is the version of the api
```

Comment:
Does Kueue care about the version? Would it work if the version wasn't set?

Reply:
I think they need version.

@dgrove-oss any ideas here on external frameworks?